data-upgrade-users/src/main/java/org/exoplatform/migration/UsersMigration.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2022 eXo Platform SAS.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.exoplatform.migration;
+
+
+import org.exoplatform.commons.search.index.IndexingService;
+import org.exoplatform.commons.upgrade.UpgradeProductPlugin;
+import org.exoplatform.commons.utils.CommonsUtils;
+import org.exoplatform.commons.utils.ListAccess;
+import org.exoplatform.container.xml.InitParams;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.services.organization.OrganizationService;
+import org.exoplatform.services.organization.User;
+import org.exoplatform.services.organization.UserStatus;
+import org.exoplatform.social.core.identity.model.Identity;
+import org.exoplatform.social.core.identity.model.Profile;
+import org.exoplatform.social.core.jpa.search.ProfileIndexingServiceConnector;
+import org.exoplatform.social.core.manager.IdentityManager;
+
+
+import java.util.Objects;
+
+public class UsersMigration extends UpgradeProductPlugin {
+    private static final Log LOG                              = ExoLogger.getExoLogger(UsersMigration.class);
+    private static final String APPLICATION_CONTENT_ID        ="migration-of-users";
+    private OrganizationService organizationService;
+    private IdentityManager identityManager;
+
+    public UsersMigration(OrganizationService organizationService,
+                          IdentityManager identityManager,
+                          InitParams initParams) {
+        super(initParams);
+        this.organizationService=organizationService;
+        this.identityManager=identityManager;
+    }
+
+    @Override
+    public void processUpgrade(String oldVersion, String newVersion) {
+        LOG.info("Start upgrade connected user profile: {}", APPLICATION_CONTENT_ID);
+        long startupTime = System.currentTimeMillis();
+        int limit=20;
+        int offset=0;
+        int totalSize = 0;
+        ListAccess<User> usersListAccess = null;
+        try {
+            usersListAccess = organizationService.getUserHandler().findAllUsers(UserStatus.ENABLED);
```

Comment:
Is there a particular reason the get only ENABLED users ?

Reply:
In fact, you should not get user like that.
When you use userManagement to search never connected users, you use the rest service UserRestResourceV1.getUsers. This function build a profileFilter and use 
        ListAccess<Identity> list = identityManager.getIdentitiesByProfileFilter(OrganizationIdentityProvider.NAME, filter, true);
to get only not connected users. 

Please use the same solution, so that you get ONLY users considered as not connected and not ALL the users.